packages/components/src/LightBox/LightBox.tsx

```diff
@@ -43,53 +47,170 @@ interface LightBoxProps {
   onRequestClose(options: RequestCloseOptions): void;
 }
 
+const swipeConfidenceThreshold = 10000;
+const swipePower = (offset: number, velocity: number) => {
+  return Math.abs(offset) * velocity;
+};
+
+const variants = {
+  enter: (direction: number) => {
+    return {
+      x: direction > 0 ? 1000 : -1000,
+      opacity: 0,
+    };
+  },
+  center: {
+    zIndex: 1,
+    x: 0,
+    opacity: 1,
+  },
+  exit: (direction: number) => {
+    return {
+      zIndex: 0,
+      x: direction < 0 ? 1000 : -1000,
+      opacity: 0,
+    };
+  },
+};
+
+const imageTransition = {
+  x: { type: "spring", stiffness: 300, damping: 30 },
+  opacity: { duration: 0.2 },
+};
+
 export function LightBox({
   open,
   images,
   imageIndex = 0,
   onRequestClose,
 }: LightBoxProps) {
   const [currentImageIndex, setCurrentImageIndex] = useState(imageIndex);
-  let nextSrc = undefined;
-  let prevSrc = undefined;
-  if (images.length > 1) {
-    nextSrc = images[(currentImageIndex + 1) % images.length].url;
-    prevSrc =
-      images[(currentImageIndex + images.length - 1) % images.length].url;
-  }
+  const [direction, setDirection] = useState(0);
+  const lightboxRef = useFocusTrap<HTMLDivElement>(open);
+
+  useRefocusOnActivator(open);
+
+  useOnKeyDown(handleRequestClose, "Escape");
+
+  useOnKeyDown(handleMovePrevious, {
+    key: "ArrowLeft",
+  });
+
+  useOnKeyDown(handleMoveNext, {
+    key: "ArrowRight",
+  });
 
   useEffect(() => {
     setCurrentImageIndex(imageIndex);
-  }, [imageIndex]);
+  }, [imageIndex, open]);
 
   return (
-    <>
+    <AnimatePresence initial={false}>
```

Comment:
we should portal this to the body since we have no control over where this get put on the component. Right now it'll render on the DOM wherever you put it on the code. The problem with that is if the parent have some `transform` property, all `position:fixed` acts like `position absolute`. Try wrapping it in a div with a `transform: translate(0)`. 

![image](https://user-images.githubusercontent.com/15986172/222744619-7361e559-3bd9-4dcf-89a7-c52fdae4bb6e.png)